Source/WebKit/WebProcess/Cache/WebCacheStorageConnection.h

```diff
@@ -63,13 +63,16 @@ class WebCacheStorageConnection final : public CanMakeWeakPtr<WebCacheStorageCon
 
     void reference(WebCore::DOMCacheIdentifier) final;
     void dereference(WebCore::DOMCacheIdentifier) final;
+    void lockStorage(const WebCore::ClientOrigin&) final;
+    void unlockStorage(const WebCore::ClientOrigin&) final;
 
     void clearMemoryRepresentation(const WebCore::ClientOrigin&, WebCore::DOMCacheEngine::CompletionCallback&&) final;
     void engineRepresentation(CompletionHandler<void(const String&)>&&) final;
     void updateQuotaBasedOnSpaceUsage(const WebCore::ClientOrigin&) final;
 
     WebCacheStorageProvider& m_provider;
     HashSet<WebCore::DOMCacheIdentifier> m_connectedIdentifiers;
+    HashMap<ClientOrigin, uint64_t> m_clientOriginLockRequestCounters;
```

Comment:
Seems like a use case for HashCountedSet?

Reply:
Will do